raft/raft.go

```diff
@@ -938,17 +1028,17 @@ func stepFollower(r *raft, m pb.Message) {
 		r.electionElapsed = 0
 		r.lead = m.From
 		r.handleSnapshot(m)
-	case pb.MsgVote:
-		if (r.Vote == None || r.Vote == m.From) && r.raftLog.isUpToDate(m.Index, m.LogTerm) {
+	case pb.MsgVote, pb.MsgPreVote:
+		if (r.Vote == None || m.Term > r.Term || r.Vote == m.From) && r.raftLog.isUpToDate(m.Index, m.LogTerm) {
```

Comment:
this m.Term > r.Term is tricky. (we need to know that this is specific for prevote since only in that case it is possible to have m.Term > r.Term)